### ISSUE TITLE
bug(MatAutocomplete): "Required Field" asterisk is clipped In High Contrast Mode

### ISSUE THREAD

**Issue by wildcardalice** [angular/components#23258]
#### Reproduction

Steps to reproduce:
1. Enable High Contrast Mode on Windows
2. Open Firefox
3. Go to demo at https://angular-dlazwc.stackblitz.io/
4. Click the input field but don't select anything, observe clipped label at top
 
#### Expected Behavior
Expected a bit more padding to the right of the asterisk to improve visibility and readability

#### Actual Behavior
The asterisk is slightly clipped when using Firefox + Windows HC (attached image is Chrome on left, Firefox on right)

![image](https://user-images.githubusercontent.com/81990996/127360223-0779adca-b566-426e-8322-f9ac3fdce92c.png)

#### Environment

- Angular: Demo forked from example at https://material.angular.io/components/autocomplete
- Browser(s): Firefox 90.0.2 (64-bit)
- Operating System: Windows 10 Enterprise 10.0.19042

**Comment by crisbeto** [comment]
I've taken a stab at this a couple of times over the years, but I haven't been able to find a good way to hide the border on Firefox in high contrast mode. For IE and Chrome we use a hack with `border-image: linear-gradient(transparent, transparent)`, but that doesn't work on Firefox for some reason. Ideally we wouldn't be using a `border` for the top spacing in the first place, but last time we tried replacing it, we couldn't quite get the baseline to align correctly.

**Comment by alicequiros** [comment]
Just commenting for reference (hopefully this helps), it looks like the MDC Web version of this component (https://github.com/material-components/material-components-web) uses a dashed border to resolve this issue.

**Comment by crisbeto** [comment]
Closing since we're now switched over to the MDC form field which doesn't seem to have this problem.
